app/database/operator/server_data_operator/handlers/group.ts

```diff
@@ -114,6 +116,75 @@ const GroupHandler = (superclass: any) => class extends superclass implements Gr
 
         return records;
     };
+
+    /**
+     * handleGroupTeamsForTeam: Handler responsible for the Create/Update operations occurring on the GroupTeam table from the 'Server' schema
+     * @param {string} teamId
+     * @param {HandleGroupTeamForTeamArgs} groupMembershipsArgs
+     * @param {GroupTeam[]} groupMembershipsArgs.groupMemberships
+     * @param {boolean} groupMembershipsArgs.prepareRecordsOnly
+     * @throws DataOperatorException
+     * @returns {Promise<GroupTeamModel[]>}
+     */
+    handleGroupTeamsForTeam = async ({teamId, groups, prepareRecordsOnly = true}: HandleGroupTeamForTeamArgs): Promise<GroupTeamModel[]> => {
+        // Get existing group memberships
+        const existingGroupTeams = await queryGroupTeamsForTeam(this.database, teamId).fetch();
+
+        let records: GroupTeamModel[] = [];
+        let rawValues: GroupTeam[] = [];
+
+        // Nothing to add or remove
+        if (!groups?.length && !existingGroupTeams.length) {
+            return records;
+        }
+
+        // No groups - remove all existing ones
```

Comment:
Perhaps my understanding is flawed and it can be corrected.
```suggestion
        // No groups - remove existing groups which are now redundant
```

Reply:
I've changed it to an if/elseif/else which should make more sense, but it removes all groups - not just reduntant ones in that if block.